packages/@expo/config-plugins/src/utils/Updates.ts

```diff
@@ -95,7 +97,16 @@ export function getRuntimeVersionNullable(
   }
 }
 
+function createFingerprint(projectRoot: string): string | null {
+  const result = child_process.execSync(`npx --yes @expo/fingerprint ${projectRoot}`, {
```

Comment:
I dont have context on why the calling methods need to be synchronous. If we can’t make this async, we could change the workflow to do the following:
- Generate the fingerprint and write the hash to a field to disk in the app config file when user runs `eas build` and `eas update` with `eas-cli` (asynchronous). 
- The plugin function that needs to be synchronous would read the newly updated `fingerprint` field from the app config 

On calling `npx`: `eas-cli` depends on this package, and it’s what everyone uses to publish updates and create builds. If we have any breaking changes, we want them to appear as users upgrade `eas-cli` versions. If we cannot apply the changes in the previous section, we can pin the version in the npx command (hacky but functional)

Reply:
> I dont have context on why the calling methods need to be synchronous

AFAIK Expo updates uses Expo's Config Plugins and config plugins are [syncronous by design.](https://docs.expo.dev/config-plugins/plugins-and-mods/)

> Generate the fingerprint and write the hash to a field to disk in the app config file when user runs eas build and eas update with eas-cli (asynchronous). At this time, we could write the fingerprint.json to disk (it could serve similar functionality as a lock file) and we could derive the hash from it.

That seems fine by me. However it does create a dependency between eas-cli and expo-fingerprint. Not sure if that makes sense from expo's perspective.